android/services/mongodb-remote/src/androidTest/java/com/mongodb/stitch/android/services/mongodb/remote/internal/SyncPerformanceTests.kt

```diff
@@ -0,0 +1,536 @@
+                if (testParams.outputToStitch) {
+                    val filterDocument = Document("_id", resultId)
+                    val updateDocument = Document()
+                        .append("\$push", Document("results", runResults.toBson()))
+                    Tasks.await(outputColl.updateOne(filterDocument, updateDocument))
+                }
+            }
+        }
+    }
+
+    private fun initialSync() {
+        val numDesiredDocs = 40
+        val documents = getDocuments(numDesiredDocs, 1024)
+        assertEquals(Tasks.await(coll.count()), 0L)
+        Tasks.await(coll.insertMany(documents))
+        assertEquals(Tasks.await(coll.count()), numDesiredDocs.toLong())
+    }
+
+    @Test
+    fun testInitialSync() {
+        val testParams = TestParams(
+            runId = runId,
+            testName = "initialSyncTest",
+            dataProbeGranularityMs = 400L,
+            docSizes = intArrayOf(1),
+            numDocs = intArrayOf(10),
+            numIters = 3,
+            numOutliersEachSide = 0,
+            outputToStitch = true
+        )
+        runPerformanceTestWithParams(testParams, this::initialSync)
+    }
+
+    private fun disconnectReconnect() {
+        val numDesiredDocs = 5000
+        val documents = getDocuments(numDesiredDocs, 1024)
+        assertEquals(Tasks.await(coll.count()), 0L)
+        Tasks.await(coll.insertMany(documents))
+        assertEquals(Tasks.await(coll.count()), numDesiredDocs.toLong())
+    }
+
+    @Test
+    fun testDisconnectReconnect() {
+        val testParams = TestParams(
+            runId = runId,
+            testName = "disconnectReconnectTest",
+            dataProbeGranularityMs = 400L,
+            docSizes = intArrayOf(1),
+            numDocs = intArrayOf(10),
+            numIters = 3,
+            numOutliersEachSide = 0,
+            outputToStitch = true
+        )
+        runPerformanceTestWithParams(testParams, this::disconnectReconnect)
+    }
+}
+
+private data class TestParams(
+    val runId: ObjectId,
+    val testName: String,
+    val numIters: Int = 12,
+    val numDocs: IntArray = intArrayOf(),
+    val docSizes: IntArray = intArrayOf(),
+    val dataProbeGranularityMs: Long = 1500L,
+    val numOutliersEachSide: Int = 1,
+    val stitchHostName: String = "",
+    val outputToStdOut: Boolean = true,
+    val outputToStitch: Boolean = false,
+    val preserveRawOutput: Boolean = false
+) {
+    fun toBson(): Document {
+        return Document()
+            .append("name", BsonString(this.testName))
+            .append("dataProbeGranularityMs", BsonInt64(this.dataProbeGranularityMs))
+            .append("numOutliersEachSide", BsonInt32(this.numOutliersEachSide))
+            .append("stitchHostName", BsonString(this.stitchHostName))
+            .append("date", BsonDateTime(Date().time))
+            .append("runId", runId)
+            .append("results", BsonArray())
+    }
+}
+
+interface DataBlock<T : Number> {
```

Comment:
[maybe] All of the subtypes can be coalesced– either add extensions to the `Number` type that allow you to do the basic maths comparisons, or use `BigDecimal`.

However, it doesn't really matter. You've already written the code and I don't think we'll be adding any other number types anytime soon. Your call.

Reply:
I trust your judgement, would like to chat about it tomorrow, but I like your idea of using BigDecimal seeing that I am doing some funky things all over the place with taking averages which returns a double and then converting that to a long which seems stupid.